### PR TITLE
Copy editing :)

### PR DESCRIPTION
**Pull request by drkameleon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ exceptions. The most notable user visible differences, which might cause Idris
 + The `cong` function now takes its congruence explicitly as its first argument.
 + Minor differences in the meaning of export modifiers `private`, `export`,
   and `public export`, which now refer to visibility of names from other
-  *namespaces* rather than visibility from other *files*.
+  *namespaces* rather than the visibility from other *files*.
```
Comment (ohad) [inline README.md:40]: Consider:
'name visibility' to clarify what visibility refers to, or
'visibility of names' to make consistent with the first part of the phrase.

File: README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Idris 2
 [![](https://github.com/idris-lang/Idris2/workflows/API/badge.svg)](https://github.com/idris-lang/Idris2/actions?query=workflow%3A"API")
 
 [Idris 2](https://idris-lang.org/) is a purely functional programming language
-with first class types.
+with first-class types.
```
Comment (ohad) [inline README.md:12]: Not an error.
OED lists the first-class adj as "usually with a hyphen", and gives a couple of examples where it is without hyphen.
The codebase currently doesn't use hyphens at all, so it's a stylistic decision we can make.
I don't have a preference.

File: README.md [hunk 3]
```diff
@@ -22,8 +22,8 @@ is often one of `scheme`, `chezscheme` or `chezscheme9.5` (depending on the
 version). On a modern desktop machine, this process (including tests)
 should take less than 5 minutes.
 
-Idris 2 is mostly backwards compatible with Idris 1, with some minor
-exceptions. The most notable user visible differences, which might cause Idris
+Idris 2 is mostly backward compatible with Idris 1, with some minor
```
Comment (ohad) [inline README.md:25]: Not an error.
According to OED, this is is still a draft addition, and usage varies, both with the suffix 's', and whether or not the adjective should be hyphenated.

File: README.md [hunk 4]
```diff
@@ -62,16 +62,16 @@ Summary of new features:
 
 + A core language based on "Quantitative Type Theory" which allows explicit
   annotation of erased types, and linear types.
-+ `let` bindings are now more expressive, and can be used to define pattern
++ `let` bindings are now more expressive, and can be used to define pattern-
   matching functions locally.
-+ Names which are in scope in a type are also always in scope in the body of
++ Names that are in scope in a type are also always in scope in the body of
```
Comment (ohad) [inline README.md:67]: Not an error, according to [Merriam-Webster](https://www.merriam-webster.com/words-at-play/when-to-use-that-and-which) (sorry for switching dictionaries, I'm not sure how to look this up in OED).
The other way around (using 'that' for non-restrictive clauses) might have been more erroneous.

File: README.md [hunk 5]
```diff
@@ -62,16 +62,16 @@ Summary of new features:
...
   the corresponding definition.
 + Better inference. Holes are global to a source file, rather than local to
   a definition, meaning that some holes can be left in function types to be
   inferred by the type checker. This also gives better inference for the types
-  of `case` expressions, and means fewer annotations are needed in interface
+  of `case` expressions and means fewer annotations are needed in interface
   declarations.
-+ Better type checker implementation which minimises the need for compile
++ Better type checker implementation which minimizes the need for compile-
```
Comment (ohad) [inline README.md:74]: Not errors.

I'm not sure what we decided about British spelling or not.

The OED only contains 'compile time' as a draft addition noun, the adjective is completely missing.

File: README.md [hunk 6]
```diff
@@ -91,7 +91,7 @@ Things still missing
 ====================
 
 + Disambiguation via 'with'
-+ Cumulativity (so we currently have Type : Type! Bear that in mind when you
++ "Cumulativity" (so we currently have Type : Type! Bear that in mind when you
```
Comment (ohad) [inline README.md:94]: Not sure why the unquoted version is an error.

**Comment by ohad** [comment]
Most of these are not errors, but we can decide on style.
The main decisions seem to be:
1. British vs. international/American spelling
2. Whether two-word adjectives should consistently be hyphenated (as opposed to their two-word nouns), e.g.:
  + at compile time (n.)
  + compile time value (adj.)
  Some two-word adjectives should always use hyphens, like 'high-level'.
3. Restricting use of 'which' to non-restrictive uses.
4. Some technical terms don't have a standard usage yet:
  + backwards compatible vs. backward compatible.
  we get to choose our convention, copy-editing should make these consistent across the code-base.

**Comment by fabianhjr** [comment]
@ohad, seems hacktoberfest related ( Many other examples on this thread: https://twitter.com/domenic/status/1311355857465225218 / https://joel.net/how-one-guy-ruined-hacktoberfest2020-drama )

**Comment by ohad** [comment]
Ah, thanks!
I wouldn't mind too much if they were real corrections though, or more substantial copy-editing.

**Comment by drkameleon** [comment]
@fabianhjr @ohad no, it's not hacktoberfest-related lol. :)

I'm a programming language designer myself and since I've - obviously - been watching several other programming-language-related projects myself, like Idris, I decided to make an initial - even though trivial - contribution.

That is all ;)

P.S. Feel free to totally ignore it if you want... I admit, the "corrections" in this case were nothing too important, but more a matter of style

**Pull request closed without merging**